### PR TITLE
Add charge input

### PR DESCRIPTION
**Pull request by YonsiG**
add charge input for usage in kernels

### CODE REVIEW

**Comment by YonsiG** [comment]
Screen Shot 2022-10-10 at 12 09 46 AM
This is the timing info. In principal, add a technical input shouldn't change anything regarding efficiency.

File: SDL/Event.cu [hunk 1]
```diff
@@ -670,6 +670,8 @@ void SDL::Event::addHitToEvent(std::vector<float> x, std::vector<float> y, std::
     // Use the actual number of hits instead of a max.
     const int nHits = x.size();
 
+//    for (int i=0; i<25; i++) printf("%f \n",x[i]);
+
```
Comment (VourMa) [inline SDL/Event.cu:674]: Maybe remove these (debugging) lines?

File: SDL/PixelTriplet.cu [hunk 2]
```diff
@@ -323,6 +323,8 @@ __device__ bool SDL::runPixelTripletDefaultAlgo(struct modules& modulesInGPU, st
         float zs[3] = {mdsInGPU.anchorZ[firstMDIndex], mdsInGPU.anchorZ[secondMDIndex], mdsInGPU.anchorZ[thirdMDIndex]};
         float rtPix[2] = {mdsInGPU.anchorRt[pixelInnerMDIndex], mdsInGPU.anchorRt[pixelOuterMDIndex]};
         float zPix[2] = {mdsInGPU.anchorZ[pixelInnerMDIndex], mdsInGPU.anchorZ[pixelOuterMDIndex]};
+        int chargePix = segmentsInGPU.charge[pixelSegmentArrayIndex];
+//        printf("%d \n ", chargePix);
```
Comment (VourMa) [inline SDL/PixelTriplet.cu:327]: Maybe remove this (debugging) line?

**Comment by YonsiG** [comment]
Thanks for the comments Manos! This is the efficiency comparison between this PR and master
https://www.classe.cornell.edu/~ygu/www/add_charge/eff_comparison_plots__2bc0096DIRTY_explicit_cache_528c4feDIRTY_explicit_cache_PU200/#1.8

**Comment by YonsiG** [comment]
I checked the nvprof and cuda-memcheck, and I saw an unfree variable :) Problem fixed

**Comment by VourMa** [comment]
Thanks for the checks, Yanxi, it looks good.

**Pull request merged**